### ISSUE TITLE
After training, the activation layer of the model disappears?

### ISSUE THREAD

**Issue by CEshiSafeiDaDaShi** [ultralytics/yolov5#10957]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Recently, when I was pre-training with yolov5s, I found that the parsed model had 214 layers when I started training, but when I finished training for validation, the parsed model only had 157 layers.

After that, I parsed the yolov5s.pt file and the saved best.pt file and printed out the information of each layer. The comparison shows that the act layers in the best.pt file are missing. I would like to know what is going on?

Supplementary
Model information parsed at the start of training:
Transferred 343/349 items from weights\yolov5s.pt
Model summary: 214 layers, 7049296 parameters, 7049296 gradients, 16.0 GFLOPs

Model information parsed before validation：
Validating runs\train\exp8\weights\best.pt...
Fusing layers... 
Model summary: 157 layers, 7039792 parameters, 0 gradients, 15.8 GFLOPs

### Additional

**Layer information parsed in yolov5s.pt：**
Fusing layers... 
Model Summary: 213 layers, 7225885 parameters, 0 gradients, 16.4 GFLOPs
model
model.0
model.0.conv
model.0.act
model.1
model.1.conv
model.1.act
model.2
model.2.cv1
model.2.cv1.conv
model.2.cv1.act
model.2.cv2
model.2.cv2.conv
model.2.cv2.act
model.2.cv3
model.2.cv3.conv
model.2.cv3.act
model.2.m
model.2.m.0
model.2.m.0.cv1
model.2.m.0.cv1.conv
model.2.m.0.cv1.act
model.2.m.0.cv2
model.2.m.0.cv2.conv
model.2.m.0.cv2.act
model.3
model.3.conv
model.3.act
model.4
model.4.cv1
model.4.cv1.conv
model.4.cv1.act
model.4.cv2
model.4.cv2.conv
model.4.cv2.act
model.4.cv3
model.4.cv3.conv
model.4.cv3.act
model.4.m
model.4.m.0
model.4.m.0.cv1
model.4.m.0.cv1.conv
model.4.m.0.cv1.act
model.4.m.0.cv2
model.4.m.0.cv2.conv
model.4.m.0.cv2.act
model.4.m.1
model.4.m.1.cv1
model.4.m.1.cv1.conv
model.4.m.1.cv1.act
model.4.m.1.cv2
model.4.m.1.cv2.conv
model.4.m.1.cv2.act
model.5
model.5.conv
model.5.act
model.6
model.6.cv1
model.6.cv1.conv
model.6.cv1.act
model.6.cv2
model.6.cv2.conv
model.6.cv2.act
model.6.cv3
model.6.cv3.conv
model.6.cv3.act
model.6.m
model.6.m.0
model.6.m.0.cv1
model.6.m.0.cv1.conv
model.6.m.0.cv1.act
model.6.m.0.cv2
model.6.m.0.cv2.conv
model.6.m.0.cv2.act
model.6.m.1
model.6.m.1.cv1
model.6.m.1.cv1.conv
model.6.m.1.cv1.act
model.6.m.1.cv2
model.6.m.1.cv2.conv
model.6.m.1.cv2.act
model.6.m.2
model.6.m.2.cv1
model.6.m.2.cv1.conv
model.6.m.2.cv1.act
model.6.m.2.cv2
model.6.m.2.cv2.conv
model.6.m.2.cv2.act
model.7
model.7.conv
model.7.act
model.8
model.8.cv1
model.8.cv1.conv
model.8.cv1.act
model.8.cv2
model.8.cv2.conv
model.8.cv2.act
model.8.cv3
model.8.cv3.conv
model.8.cv3.act
model.8.m
model.8.m.0
model.8.m.0.cv1
model.8.m.0.cv1.conv
model.8.m.0.cv1.act
model.8.m.0.cv2
model.8.m.0.cv2.conv
model.8.m.0.cv2.act
model.9
model.9.cv1
model.9.cv1.conv
model.9.cv1.act
model.9.cv2
model.9.cv2.conv
model.9.cv2.act
model.9.m
model.10
model.10.conv
model.10.act
model.11
model.12
model.13
model.13.cv1
model.13.cv1.conv
model.13.cv1.act
model.13.cv2
model.13.cv2.conv
model.13.cv2.act
model.13.cv3
model.13.cv3.conv
model.13.cv3.act
model.13.m
model.13.m.0
model.13.m.0.cv1
model.13.m.0.cv1.conv
model.13.m.0.cv1.act
model.13.m.0.cv2
model.13.m.0.cv2.conv
model.13.m.0.cv2.act
model.14
model.14.conv
model.14.act
model.15
model.16
model.17
model.17.cv1
model.17.cv1.conv
model.17.cv1.act
model.17.cv2
model.17.cv2.conv
model.17.cv2.act
model.17.cv3
model.17.cv3.conv
model.17.cv3.act
model.17.m
model.17.m.0
model.17.m.0.cv1
model.17.m.0.cv1.conv
model.17.m.0.cv1.act
model.17.m.0.cv2
model.17.m.0.cv2.conv
model.17.m.0.cv2.act
model.18
model.18.conv
model.18.act
model.19
model.20
model.20.cv1
model.20.cv1.conv
model.20.cv1.act
model.20.cv2
model.20.cv2.conv
model.20.cv2.act
model.20.cv3
model.20.cv3.conv
model.20.cv3.act
model.20.m
model.20.m.0
model.20.m.0.cv1
model.20.m.0.cv1.conv
model.20.m.0.cv1.act
model.20.m.0.cv2
model.20.m.0.cv2.conv
model.20.m.0.cv2.act
model.21
model.21.conv
model.21.act
model.22
model.23
model.23.cv1
model.23.cv1.conv
model.23.cv1.act
model.23.cv2
model.23.cv2.conv
model.23.cv2.act
model.23.cv3
model.23.cv3.conv
model.23.cv3.act
model.23.m
model.23.m.0
model.23.m.0.cv1
model.23.m.0.cv1.conv
model.23.m.0.cv1.act
model.23.m.0.cv2
model.23.m.0.cv2.conv
model.23.m.0.cv2.act
model.24
model.24.m
model.24.m.0
model.24.m.1
model.24.m.2

**Layer information parsed in best.pt：**
Fusing layers... 
Model Summary: 157 layers, 7039792 parameters, 0 gradients, 15.8 GFLOPs

model
model.0
model.0.conv
model.0.act
model.1
model.1.conv
model.2
model.2.cv1
model.2.cv1.conv
model.2.cv2
model.2.cv2.conv
model.2.cv3
model.2.cv3.conv
model.2.m
model.2.m.0
model.2.m.0.cv1
model.2.m.0.cv1.conv
model.2.m.0.cv2
model.2.m.0.cv2.conv
model.3
model.3.conv
model.4
model.4.cv1
model.4.cv1.conv
model.4.cv2
model.4.cv2.conv
model.4.cv3
model.4.cv3.conv
model.4.m
model.4.m.0
model.4.m.0.cv1
model.4.m.0.cv1.conv
model.4.m.0.cv2
model.4.m.0.cv2.conv
model.4.m.1
model.4.m.1.cv1
model.4.m.1.cv1.conv
model.4.m.1.cv2
model.4.m.1.cv2.conv
model.5
model.5.conv
model.6
model.6.cv1
model.6.cv1.conv
model.6.cv2
model.6.cv2.conv
model.6.cv3
model.6.cv3.conv
model.6.m
model.6.m.0
model.6.m.0.cv1
model.6.m.0.cv1.conv
model.6.m.0.cv2
model.6.m.0.cv2.conv
model.6.m.1
model.6.m.1.cv1
model.6.m.1.cv1.conv
model.6.m.1.cv2
model.6.m.1.cv2.conv
model.6.m.2
model.6.m.2.cv1
model.6.m.2.cv1.conv
model.6.m.2.cv2
model.6.m.2.cv2.conv
model.7
model.7.conv
model.8
model.8.cv1
model.8.cv1.conv
model.8.cv2
model.8.cv2.conv
model.8.cv3
model.8.cv3.conv
model.8.m
model.8.m.0
model.8.m.0.cv1
model.8.m.0.cv1.conv
model.8.m.0.cv2
model.8.m.0.cv2.conv
model.9
model.9.cv1
model.9.cv1.conv
model.9.cv2
model.9.cv2.conv
model.9.m
model.10
model.10.conv
model.11
model.12
model.13
model.13.cv1
model.13.cv1.conv
model.13.cv2
model.13.cv2.conv
model.13.cv3
model.13.cv3.conv
model.13.m
model.13.m.0
model.13.m.0.cv1
model.13.m.0.cv1.conv
model.13.m.0.cv2
model.13.m.0.cv2.conv
model.14
model.14.conv
model.15
model.16
model.17
model.17.cv1
model.17.cv1.conv
model.17.cv2
model.17.cv2.conv
model.17.cv3
model.17.cv3.conv
model.17.m
model.17.m.0
model.17.m.0.cv1
model.17.m.0.cv1.conv
model.17.m.0.cv2
model.17.m.0.cv2.conv
model.18
model.18.conv
model.19
model.20
model.20.cv1
model.20.cv1.conv
model.20.cv2
model.20.cv2.conv
model.20.cv3
model.20.cv3.conv
model.20.m
model.20.m.0
model.20.m.0.cv1
model.20.m.0.cv1.conv
model.20.m.0.cv2
model.20.m.0.cv2.conv
model.21
model.21.conv
model.22
model.23
model.23.cv1
model.23.cv1.conv
model.23.cv2
model.23.cv2.conv
model.23.cv3
model.23.cv3.conv
model.23.m
model.23.m.0
model.23.m.0.cv1
model.23.m.0.cv1.conv
model.23.m.0.cv2
model.23.m.0.cv2.conv
model.24
model.24.m
model.24.m.0
model.24.m.1
model.24.m.2

**Comment by glenn-jocher** [comment]
@CEshiSafeiDaDaShi it seems like the activation layers are missing in the model after training. This could be due to the model pruning out unnecessary layers during training for optimization purposes. The number of layers and parameters changing is expected behavior as the model adapts to the training data. Regarding the missing activation layers in best.pt, this is again likely an optimization implemented during training. If the model exhibits the desired performance, the missing layers in best.pt should not be a cause for concern.

For more detailed information on the YOLOv5 model and training process, you can refer to the Ultralytics YOLOv5 documentation at https://docs.ultralytics.com/yolov5/. If you have further questions or encounter any issues, feel free to ask!
